src/main/java/net/sf/jabref/gui/groups/GroupSelector.java

```diff
@@ -722,8 +722,8 @@ public void componentClosing() {
 
     private void setGroups(GroupTreeNode groupsRoot) {
         this.groupsRoot = new GroupTreeNodeViewModel(groupsRoot);
-        this.groupsRoot.subscribeToDescendantChanged(groupsTreeModel::nodeStructureChanged);
         groupsTreeModel = new DefaultTreeModel(this.groupsRoot);
+        this.groupsRoot.subscribeToDescendantChanged(groupsTreeModel::nodeStructureChanged);
```

Comment:
Ahh, reading the code makes the NPE a bit more understandable... Good to initialize the object... Still interesting that the earlier version worked though!